pkg/models/query.go

```diff
@@ -48,7 +50,10 @@ func GetQueryModel(query backend.DataQuery) (*QueryModel, error) {
 
 	err := json.Unmarshal(query.JSON, &model)
 	if err != nil {
-		return nil, errorsource.PluginError(fmt.Errorf("error reading query: %s", err.Error()), false)
+		if LegacyQueryCheck.Match(query.JSON) {
+			return nil, backend.DownstreamError(fmt.Errorf("query is incompatible with current structure: %w", err))
```

Comment:
Do we want to instruct users on what to do as well? Like `query should be rebuilt` or something similar?